tianshou/policy/modelfree/sac.py

```diff
@@ -125,6 +130,9 @@ def forward(  # type: ignore
         if self._noise is not None and not self.updating:
             act += to_torch_as(self._noise(act.shape), act)
         act = act.clamp(self._range[0], self._range[1])
+        if self._deterministic_eval and not self.training:
+            act = torch.tanh(logits[0]) * \
+                self._action_scale + self._action_bias
```

Comment:
Stochastic action is sampled before these lines causing the `log_prob` to be wrong and `act.clamp` to be ignored. Suggest coding like `x = logits[0] if self._deterministic_eval and not self.training else dist.rsample()` and updating the logic of adding noise instead.